tests/lax_test.py

```diff
@@ -62,6 +62,18 @@
 
 compatible_shapes = [[(3,)], [(3, 4), (3, 1), (1, 4)], [(2, 3, 4), (2, 1, 4)]]
 
+# We check cases where the preferred type is at least as wide as the input
+# type and where both are either both floating-point or both integral,
+# which are the only supported configurations.
+preferred_type_combinations = [
+  (np.float16, np.float16), (np.float16, np.float32), (np.float16, np.float64),
+  (dtypes.bfloat16, dtypes.bfloat16), (dtypes.bfloat16, np.float32),
+  (dtypes.bfloat16, np.float64), (np.float32, np.float32), (np.float32, np.float64),
+  (np.float64, np.float64), (np.int8, np.int8), (np.int8, np.int16), (np.int8, np.int32),
+  (np.int8, np.int64), (np.int16, np.int16), (np.int16, np.int32), (np.int16, np.int64),
+  (np.int32, np.int32), (np.int32, np.int64), (np.int64, np.int64),
+  (np.complex64, np.complex64), (np.complex64, np.complex128), (np.complex128, np.complex128)]
```

Comment:
`np.complex128` is currently not supported on TPU, so we need to avoid this test case for that device (you can check via `jtu.device_under_test() == 'tpu'`)

Reply:
Also on CPU we're seeing some failures due to insufficient precision:
```
LaxTest.testConvPreferredElement_lhs_shape=complex64[2,2,9,10]_rhs_shape=complex64[3,2,4,5]_preferred_element_type=<class 'numpy.complex128'>
```
You can probably reproduce this locally with
```
$ JAX_NUM_GENERATED_CASES=50 pytest -n auto tests/lax_test.py -k PreferredElement
```
Also worth testing with X64 enabled:
```
$ JAX_ENABLE_X64=1 JAX_NUM_GENERATED_CASES=50 pytest -n auto tests/lax_test.py -k PreferredElement
```

Reply:
Thanks for checking internal CI. I skipped the test on TPU and reduced the precision on CPU for the mentioned cases which should fix the failures.